torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -464,7 +464,12 @@ class CudaKernelGenerator : private OptOutConstDispatch {
     } else if (ns->definition() == nullptr) {
       code_ << ns->name();
     } else {
-      code_ << genInline(ns->definition());
+      const bool has_alloc = alloc_map_.find(ns) != alloc_map_.end();
```

Comment:
Not for this PR, but it seems possible to clean up the `handle` functions for scalars as they seem mostly follow the same logic.